install/install.go

```diff
@@ -153,17 +198,49 @@ func (i *Installer) InstallArgoCDServer() {
 	argoCDServerControllerDeployment.Spec.Template.Spec.InitContainers[0].ImagePullPolicy = apiv1.PullPolicy(i.ImagePullPolicy)
 	argoCDServerControllerDeployment.Spec.Template.Spec.Containers[0].Image = i.ServerImage
 	argoCDServerControllerDeployment.Spec.Template.Spec.Containers[0].ImagePullPolicy = apiv1.PullPolicy(i.ImagePullPolicy)
+
+	rootCredentialsSecretName := util.ConfigManagerDefaultRootCredentialsSecretName
 	// Use a Kubernetes ConfigMap, if provided.
 	if i.InstallOptions.ConfigMap != "" {
-		configMap := strconv.Quote(i.InstallOptions.ConfigMap)
+
+		// Used for reading config maps and secrets here
+		kubeclientset, err := kubernetes.NewForConfig(i.config)
+		errors.CheckError(err)
+
+		quotedConfigMapName := strconv.Quote(i.InstallOptions.ConfigMap)
 		container := &argoCDServerControllerDeployment.Spec.Template.Spec.Containers[0]
-		container.Command = append(container.Command, "--config-map", configMap)
+		container.Command = append(container.Command, "--config-map", quotedConfigMapName)
+		configMap, err := kubeclientset.CoreV1().ConfigMaps(i.Namespace).Get(i.InstallOptions.ConfigMap, metav1.GetOptions{})
+		errors.CheckError(err)
+
+		secretNameOverride, ok := configMap.Data[util.RootCredentialsSecretNameKey]
```

Comment:
Sorry for asking to make changes again :) . Even if configMap is not supplied then configMap with default name might exist and it might have non-default admin secret name. If user requested to create secret with password installer needs to read either default config map or config map with the provided name. If map does not exist, then default secret name might be used.

Reply:
@alexmt to resolve this, I could see either just relying on the default secret name (ignoring the possibility of a default config map), or I could see us adding a default config map feature.  I'm concerned that if we take the second route, it might actually go more smoothly if that's a follow up feature after some careful planning, because we have design decisions to make then (do we insist on there always being a default config map? when do we create it? for what is it consulted?).  Interested in your thoughts!